notebooks/oracle2bq/OracleToBigQuery_notebook.ipynb

```diff
@@ -0,0 +1,888 @@
+    " - NUMBER_OF_PARTITIONS : Number of partitioned reads default 10\n",
+    " - CHECK_PARTITION_COLUMN_LIST : List will have table and its partitioned column if exceeds threshold"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "fe54e83c-7845-4043-baf8-c0ee6f32787a",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "PARTITION_THRESHOLD = 100000\n",
+    "CHECK_PARTITION_COLUMN_LIST={}\n",
+    "NUMBER_OF_PARTITIONS = 10"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "b692a051-7555-42fc-af27-b05e461576f4",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "with DB.connect() as conn:\n",
+    "    for table in ORACLETABLE_LIST:\n",
+    "        results = DB.execute(\"SELECT count(1) FROM {}\".format(table)).fetchall()\n",
+    "        if results[0][0]>int(PARTITION_THRESHOLD) and len(SQL_TABLE_PRIMARY_KEYS.get(table).split(\",\")[0])>0:\n",
+    "            column_list=SQL_TABLE_PRIMARY_KEYS.get(table).split(\",\")\n",
+    "            for column in column_list:\n",
+    "                print(column)\n",
+    "                results_datatype = DB.execute(\"select DATA_TYPE from sys.all_tab_columns where TABLE_NAME = '{0}' and COLUMN_NAME = '{1}'\".format(table,column)).fetchall()      \n",
+    "                print(results_datatype[0][0])\n",
+    "                if results_datatype[0][0]==\"NUMBER\":\n",
+    "                    lowerbound = DB.execute(\"SELECT min({0}) from {1}\".format(column,table)).fetchall()\n",
+    "                    upperbound = DB.execute(\"SELECT max({0}) from {1}\".format(column,table)).fetchall()\n",
+    "                    CHECK_PARTITION_COLUMN_LIST[table]=[column,lowerbound[0][0],upperbound[0][0]]                \n",
+    "print(CHECK_PARTITION_COLUMN_LIST)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "1fa5f841-a687-4723-a8e6-6e7e752ba36e",
+   "metadata": {},
+   "source": [
+    "## Step 7: Download JAR files and Upload to GCS (only rquired to run one-time)\n",
+    "#### Run Step 7 one time for each new notebook instance"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "22220ae3-9fb4-471c-b5aa-f606deeca15e",
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "%cd $WORKING_DIRECTORY"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "bdee7afc-699b-4c1a-aeec-df0f99764ae0",
+   "metadata": {},
+   "source": [
+    "#### Downloading JDBC Oracle Driver and Bigquery Spark Connector Jar files"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "4b40f634-1983-4267-a4c1-b072bf6d81ae",
+   "metadata": {
+    "scrolled": true,
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "%%bash\n",
+    "wget https://repo1.maven.org/maven2/com/oracle/database/jdbc/ojdbc8/21.7.0.0/ojdbc8-21.7.0.0.jar\n",
```

Comment:
Download only when jar is missing.

Reply:
was getting error without them, only required one downloaded